artifactory.py

```diff
@@ -1015,14 +1037,27 @@ def del_properties(self, pathobj, props, recursive):
     def scandir(self, pathobj):
         return _ScandirIter((pathobj.joinpath(x) for x in self.listdir(pathobj)))
 
-    def writeto(self, fd, out, chunk_size):
-        url = str(fd)
-        res = fd.session.get(url, stream=True, verify=True, cert=None)
-        if res.status_code != 200:
-            raise RuntimeError(res.status_code)
-        for chunk in res.iter_content(chunk_size=chunk_size):
-            if chunk:
-                out.write(chunk)
+    def writeto(
+        self, pathobj, file, chunk_size=1024, progress_func=print_download_progress
+    ):
+        """
+        Downloads large file in chunks and prints progress
+        :param pathobj: path like object
+        :param file: IO object
+        :param chunk_size: chunk size in bytes, recommend. eg 1024*1024 is 1Mb
+        :param progress_func: by default print_download_progress. Provide custom function to print output or suppress
```

Comment:
`by default print_download_progress` - we can see it at the defaults, no needs to document the name, I suppose. Default behavior - we can document it, but not the variable names.